src/fmi4c.c

```diff
@@ -1601,53 +1596,56 @@ bool loadFunctionsFmi1(fmiHandle *fmu)
 
     bool ok = true;
 
+    // The temp buffer is used for concatenating modelnName_functionName to avoid memory if such a buffer would be allocated inside getFunctionName
+    char tmpbuff[FILENAME_MAX];
```

Comment:
Ok, probably a good idea. It makes the lines below a little more complicated, but I can't think of a better solution right now.

Reply:
yes its a bit ugly to assume we know the size, but since `getFunctionName` is an internal function I think its OK, otherwise we could supply the size as an argument, Also I chose not to use an internal buffer since it would have to be static and would cause problem if multiple threads are calling it at the same time.